apps/web/pages/signup.tsx

```diff
@@ -0,0 +1,194 @@
+import type { GetServerSidePropsContext } from "next";
+import { signIn } from "next-auth/react";
+import { useRouter } from "next/router";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+
+import LicenseRequired from "@calcom/features/ee/common/components/v2/LicenseRequired";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { inferSSRProps } from "@calcom/types/inferSSRProps";
+import { Alert } from "@calcom/ui/Alert";
+import Button from "@calcom/ui/v2/core/Button";
+import { EmailField, PasswordField, TextField } from "@calcom/ui/v2/core/form/fields";
+import { HeadSeo } from "@calcom/web/components/seo/head-seo";
+import { asStringOrNull } from "@calcom/web/lib/asStringOrNull";
+import { WEBAPP_URL } from "@calcom/web/lib/config/constants";
+import prisma from "@calcom/web/lib/prisma";
+import { isSAMLLoginEnabled } from "@calcom/web/lib/saml";
+import { IS_GOOGLE_LOGIN_ENABLED } from "@calcom/web/server/lib/constants";
+import { ssrInit } from "@calcom/web/server/lib/ssr";
+
+type FormValues = {
+  username: string;
+  email: string;
+  password: string;
+  passwordcheck: string;
+  apiError: string;
+};
+
+export default function Signup({ prepopulateFormValues }: inferSSRProps<typeof getServerSideProps>) {
+  const { t } = useLocale();
+  const router = useRouter();
+  const methods = useForm<FormValues>({
+    defaultValues: prepopulateFormValues,
+  });
+  const {
+    register,
+    formState: { errors, isSubmitting },
+  } = methods;
+
+  const handleErrors = async (resp: Response) => {
+    if (!resp.ok) {
+      const err = await resp.json();
+      throw new Error(err.message);
+    }
+  };
+
+  const signUp: SubmitHandler<FormValues> = async (data) => {
+    await fetch("/api/auth/signup", {
+      body: JSON.stringify({
+        ...data,
+      }),
+      headers: {
+        "Content-Type": "application/json",
+      },
+      method: "POST",
+    })
+      .then(handleErrors)
+      .then(
+        async () =>
+          await signIn("Cal.com", {
+            callbackUrl: (`${WEBAPP_URL}/${router.query.callbackUrl}` || "") as string,
+          })
+      )
+      .catch((err) => {
+        methods.setError("apiError", { message: err.message });
+      });
+  };
+
+  return (
+    <LicenseRequired>
```

Comment:
Two extra cents here @emrysal @PeerRich This page is mostly going to be accessed publicly and LicenseRequired needs a logged in user in order to work. So all unauthed users will se the "license required" screen.